addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -0,0 +1,264 @@
+
+        squeezeBoxServerHandler.registerSqueezeBoxPlayerListener(listener);
+        squeezeBoxServerHandler.deletePlaylistItem(mac, notificationMessagePlaylistsIndex);
+
+        try {
+            waitForPlaylistUpdate(listener);
+        } finally {
+            squeezeBoxServerHandler.unregisterSqueezeBoxPlayerListener(listener);
+        }
+    }
+
+    /**
+     * Monitor the number of playlist entries. When it changes, then we know the playlist
+     * has been updated with the notification URL. There's probably an edge case here where
+     * someone is updating the playlist at the same time, but that should be rare.
+     *
+     * @param listener
+     * @throws InterruptedException
+     * @throws SqueezeBoxTimeoutException
+     */
+    private void waitForPlaylistUpdate(SqueezeBoxNotificationListener listener)
+            throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.trace("Waiting up to {} s for playlist to be updated...", PLAYLIST_COMMAND_TIMEOUT);
+
+        int timeoutCount = 0;
+
+        while (!listener.isPlaylistUpdated()) {
+            Thread.sleep(100);
+            if (timeoutCount++ > PLAYLIST_COMMAND_TIMEOUT * 10) {
+                throw new SqueezeBoxTimeoutException("Unable to update playlist.");
+            }
+        }
+
+        this.playlistModified = true;
+    }
+
+    private void playNotification() throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.debug("Playing notification");
+
+        notificationMessagePlaylistsIndex = squeezeBoxPlayerHandler.currentNumberPlaylistTracks() - 1;
+        SqueezeBoxNotificationListener listener = new SqueezeBoxNotificationListener(mac);
+        listener.resetStopped();
+
+        squeezeBoxServerHandler.registerSqueezeBoxPlayerListener(listener);
+        squeezeBoxServerHandler.playPlaylistItem(mac, notificationMessagePlaylistsIndex);
+
+        logger.trace("Waiting up to {} s for stop...", MAX_NOTIFICATION_LENGTH);
+
+        try {
+            int timeoutCount = 0;
+
+            while (!listener.isStopped()) {
+                Thread.sleep(100);
+                if (timeoutCount++ > MAX_NOTIFICATION_LENGTH * 10) {
+                    throw new SqueezeBoxTimeoutException("Unable to play message.");
+                }
+            }
+        } finally {
+            squeezeBoxServerHandler.unregisterSqueezeBoxPlayerListener(listener);
+        }
+    }
+
+    private void restorePlayerState() {
+        logger.debug("Restoring player state");
+
+        // Resume playing save playlist item if player wasn't stopped.
+        // Note that setting the time doesn't work for remote streams.
+        if (!playerState.isStopped()) {
+            logger.debug("Resuming last item playing");
+            squeezeBoxServerHandler.playPlaylistItem(mac, playerState.getPlaylistIndex());
+            squeezeBoxServerHandler.setPlayingTime(mac, playerState.getPlayingTime());
+        } else if (!playerState.isPlaying()) {
```

Comment:
@pgfeller There is a possibly that a condition is not handled here. I'm thinking of the case where the saved player state is not stopped (isStopped == false) and is paused (isPlaying == false). In the above code, the pause will not be sent if the player is not stopped.